framework/infrastructure/traffic_director.py

```diff
@@ -593,8 +605,12 @@ def delete_alternative_url_map(self, force=False):
         self.compute.delete_url_map(name)
         self.url_map = None
 
-    def create_target_proxy(self):
-        name = self.make_resource_name(self.TARGET_PROXY_NAME)
+    def create_target_proxy(self, ipv6=False):    
+        if ipv6:
+            name = self.make_resource_name(self.TARGET_PROXY_NAME + "-v6")
```

Comment:
feel free to add `TARGET_PROXY_NAME_IPV6` constant (similar to `FIREWALL_RULE_NAME_IPV6` we already have) to avoid doing `+ "-v6"`  all the time